torch/csrc/distributed/autograd/context/dist_autograd_context.cpp

```diff
@@ -94,10 +114,35 @@ void DistAutogradContext::setGraphTask(
 
 void DistAutogradContext::addOutstandingRpc(
     const std::shared_ptr<rpc::FutureMessage>& futureMessage) {
+  futureMessage->addCallback(std::bind(
+      &DistAutogradContext::outStandingRpcCallback,
+      this,
+      std::placeholders::_1));
   std::lock_guard<std::mutex> guard(lock_);
   outStandingRpcs_.push_back(futureMessage);
 }
 
+void DistAutogradContext::outStandingRpcCallback(const rpc::Message& message) {
+  if (message.type() == rpc::MessageType::EXCEPTION) {
+    // If we have an error, let the local autograd engine know about it.
+    std::string err(message.payload().begin(), message.payload().end());
+    auto exception = std::runtime_error(err);
+
+    // Enqueue 'ErrorFunc' on the local autograd engine.
+    auto& localEngine = torch::autograd::Engine::get_default_engine();
+    auto errorFunc = std::make_shared<ErrorFunc>(exception);
+
+    // Increment out standing tasks for this function and set appropriate
+    // exec_info_ for this function to execute.
+    graphTask_->outstanding_tasks_++;
```

Comment:
I am wondering if it is possible for `graphTask_->outstanding_tasks_ == 0` here. Whether or not this is possible depends on the relationship between outstanding rpcs and outstanding tasks. Can this invariant get clarified here?

Reply:
I don't think we need the logic here if we just set an error on the graphTask.